src/types.rs

```diff
@@ -0,0 +1,624 @@
+    /// [`EncryptAlgorithm::UnsafeTrippleDes2Tdea`]: enum.EncryptAlgorithm.html#variant.UnsafeTrippleDes2Tdea
+    /// [`64`]: enum.KeyLength.html#variant.Bits
+    /// [`128`]: enum.KeyLength.html#variant.Bits
+    /// [`192`]: enum.KeyLength.html#variant.Bits
+    /// [`256`]: enum.KeyLength.html#variant.Bits
+    /// [`DES`]: enum.KeyType.html#variant.Des
+    Wrap,
+}
+
+// Used by Context property functions
+#[derive(Debug, PartialEq)]
+pub(crate) enum Property {
+    Padding,
+    GcmAad,
+    GcmTag,
+    GcmTagLen,
+    CcmAad,
+    CcmTag,
+    CcmTagLen,
+    Rc2EffectiveKeyBits,
+}
+
+/// Enumeration of YACA paddings
+#[derive(Debug, PartialEq)]
+pub enum Padding {
+    /// No padding at all. This method assumes that the input data
+    /// already has a proper length for a given cryptographic
+    /// operation (e.g. it has been padded by the client). Suitable
+    /// for symmetric `Encrypt`/`Decrypt` operations as well as low-level
+    /// `RSA` operations.
+    None,
+
+    /// X9.31 padding. Suitable for `RSA` `Sign`/`Verify` operation. Not
+    /// supported in low-level `RSA` operations.
+    X931,
+
+    /// PKCS #1 v1.5 padding. Suitable for `RSA` `Sign`/`Verify` and
+    /// low-level `RSA` operations. For low-level operations the input
+    /// must be at least 11 bytes shorter than the key length.
+    Pkcs1,
+
+    /// PKCS #1 PSS padding. Suitable for `RSA` `Sign`/`Verify`
+    /// operations. Not supported in low-level `RSA` operations.
+    Pkcs1Pss,
+
+    /// EME-OAEP as defined in PKCS #1 v2.0 with SHA-1, MGF1 and an
+    /// empty encoding parameter. Suitable for low-level `RSA`
+    /// public_encrypt/private_decrypt operations. For low-level
+    /// operations the input must be at least 42 bytes shorter than
+    /// the key length.
+    Pkcs1Oaep,
+
+    /// PKCS #1 v1.5 padding with an SSL-specific modification that
+    /// denotes that the party is SSL3 capable. It is used for
+    /// rollback attack detection in SSLv3. If during decryption it
+    /// turns out that both parties are using `Pkcs1SslV23` (both are
+    /// communicating using SSL2 and both are SSL3 capable) it is
+    /// treated as a rollback attack and an error is
+    /// returned. Suitable for low-level `RSA`
+    /// `public_encrypt`/`private_decrypt` operations. For low-level
+    /// operations the input must be at least 11 bytes shorter than
+    /// the key length.
+    Pkcs1SslV23,
+
+    /// PKCS #7 padding. Suitable for symmetric `Encrypt`/`Decrypt` operation.
+    Pkcs7,
+}
+
+/// Enumeration of YACA key derivation functions
+#[derive(Debug, PartialEq)]
+pub enum Kdf {
+    /// ANSI X9.42 key derivation function, (shared secret derived
+    /// using Diffie-Helmann key exchange protocol).
+    X942,
+
+    /// ANSI X9.62 key derivation function, (shared secret derived
+    /// using EC Diffie-Helmann key exchange protocol).
+    X962,
+}
+
+// Local Variables:
```

Comment:
For my emacs not to remove trailing whitespaces at the end of lines in this file. Can't you read? ;-P

Reply:
You may have forgotten the f word.

Reply:
Would it be clearer if I hadn't? ;-)

Reply:
Talking seriously:
https://www.gnu.org/software/emacs/manual/html_node/emacs/Specifying-File-Variables.html#Specifying-File-Variables

This is a normal thing for emacs users. It doesn't hurt anyone. 

Trailing whitespaces are bad. But for reasons unknown (probably some brainfart) markdown uses them. If anyone it's them that deserve the f word.

Reply:
Just because one has a pedal doesn't mean everyone & their mother have to look at it ;)

In all seriousness though, why just the one file?

Reply:
Because it was the only file where I had (wanted) to use markdown's newline marker. This file is heavy with rustdoc. Vim users also mark files in similar way sometimes. I don't mind.

Reply:
Ack.